src/Analyzer.Cli/CommandLineParser.cs

```diff
@@ -235,15 +241,30 @@ private void AnalyzeDirectory(DirectoryInfo directoryPath, ReportFormat reportFo
                         {
                             logger.LogError("\t{failedFile}", failedFile);
                         }
+                        return 6;
                     }
+                    return exitCode;
                 }
             }
             catch (Exception exp)
             {
                 logger.LogError(GetExceptionMessage(exp));
+                return 1;
             }
         }
 
+        private int CalculateExitCode(int exitCode, int res)
```

Comment:
I think the logic here can be greatly simplified, to the point of maybe not needing this function.  For example, since this function is only called from `AnalyzeDirectory()`, `res` will never be anything other than 0, 1, or 4.  Nothing different is needed for 4, and whether 0s and 1s are returned can be detected at the end by looking at `numOfSuccesses` and `filesFailed.Count`.

Reply:
Oh, I thought 2 and 3 could happen in `AnalyzeTemplate()`...

Reply:
2 and 3 can be returned from that function, but when called from `AnalyzeDirectory()`, they never will be.
- 2 means the file doesn't exist - but the file will always exist in this case because it was discovered from `FindJsonFilesInDirectoryRecursive()`
- 3 will never be returned because `writer` is always passed from `AnalyzeDirectory()`

Reply:
Thank you for clarifying Johnathon!

Reply:
Wait so what about when we add in 5? Then would the logic be needed?

Reply:
Where is 5 initially returned/generated? I don't see it.

Reply:
Yes, that is one of my questions

Reply:
Was `5` never implemented in #229?

Reply:
Can it also be simplified by making them different enums (e.g. a set of enums of errors, issues, and success) or something? I'm thinking some other type of data structure where you have more to compare than just ints.

Reply:
One option is to do bitwise enums and `|` values together to form the final exit code.

Reply:
Revisiting no longer needed bitwise enums and also thought it would be better for the user that exit codes are sequential